catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/matchers/FormulaEditorCategoryListMatchers.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.util.matchers;
+
+import android.view.View;
+import android.widget.ListView;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.adapter.CategoryListAdapter;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+public final class FormulaEditorCategoryListMatchers {
+	private FormulaEditorCategoryListMatchers() {
+		throw new AssertionError();
+	}
+
+	public static Matcher<View> isFormulaEditorCategoryListView() {
+		return new TypeSafeMatcher<View>() {
+
+			@Override
+			protected boolean matchesSafely(View view) {
+				return view instanceof ListView && ((ListView) view).getAdapter() instanceof CategoryListAdapter;
+			}
+
+			@Override
+			public void describeTo(Description description) {
+				description.appendText("FormulaEditorCategoryListView");
+			}
+		};
+	}
+
+	public static Matcher<View> isFormulaEditorListItem() {
+		return new TypeSafeMatcher<View>() {
+			@Override
+			protected boolean matchesSafely(View item) {
+				return item.getId() == R.id.fragment_formula_editor_list_item;
```

Comment:
why not user withId(R.id.fragment_formula_editor_list_item) since this is only used once?